---
title: "Source data Extended Data Fig. 7d and 9a. Legends Extended Data Fig. 7d and 9a"
author: "Marcos Morgan"
output: html_document
---

Load library
```{r, message=FALSE, warning=FALSE}
library(dplyr)
library(tidyr)
library(knitr)
library(ggplot2)
```

Clean the environment
```{r, message=FALSE, warning=FALSE}
rm(list=ls()[!ls() %in% c("table")])
```

Set working directory
```{r setup}
knitr::opts_chunk$set(warning = FALSE, message = FALSE)
opts_knit$set(root.dir = "../")
```

Load helper functions
```{r, message=FALSE, warning=FALSE}
source("scripts/figures_helper_functions.R")
```

Read table
```{r, message=FALSE, warning=FALSE}

if(!(c("table") %in% ls())){
  
  #Loading table
  table <- read.csv("views/table/TAIL-seq_Morganetal2017.csv", stringsAsFactors=FALSE)
  table <- table %>% rename(palen = poly_a_tail_length, cell = cell_tissue, mod = modification, short = short_tail, 
                            type = modification_type, type_short = modification_type_short, gv = gv_upregulated) %>% 
                     rowwise() %>% mutate(cell      = rename_cell(cell)) %>%
                     rowwise() %>% mutate(treatment = rename_treatment(cell, treatment)) %>% ungroup()
  
} 

head(table)
```

Generate table for modification types without dividing by mono- and oligo-.
```{r, message=FALSE, warning=FALSE}
reads_stats   <- table %>% group_by(cell, treatment, sample, short) %>% mutate(all       = n()) %>% ungroup() %>%
                    group_by(cell, treatment, sample, short, type_short) %>% mutate(mod_count = n()) %>% ungroup() %>%
                    rowwise() %>% mutate(frequency = 100*mod_count/all)                                            %>%
                    select(cell, treatment, sample, short, type_short, frequency) %>% unique()         

reads_stats   <- missing_T(reads_stats)

reads_stats   <- reads_stats %>% group_by(cell, treatment, short, type_short) %>%
                    mutate(mean = mean(frequency), sd = sd(frequency), frequencies=list(frequency)) %>% ungroup()

reads_stats     <- subset(reads_stats , !is.na(reads_stats$sd))
show(reads_stats)
```

```{r, message=FALSE, warning=FALSE}
reads_stats_t   <- reads_stats %>% select(cell, treatment, short, type_short, frequencies) %>% unique() %>% spread(treatment, frequencies)
reads_stats_t   <- reads_stats_t %>% rowwise() %>% 
                                     mutate( pval = ifelse(length(dcko) > 1  & length(ctrl) > 1, 
                                                             t.test(dcko, ctrl, var.equal=TRUE)$p.value, NA)) %>%
                                     mutate( star = ifelse(!is.na(pval), sig(pval),NA)) %>%
                                     select(cell, short, type_short, pval, star)

reads_stats   <- full_join(reads_stats, reads_stats_t) %>% select(-frequencies)
show(reads_stats)

reads_stats_f   <- reads_stats %>% select(cell, treatment, short, type_short, mean) %>% unique() %>% spread(treatment, mean)
reads_stats_f   <- reads_stats_f %>% rowwise() %>% 
                                     mutate(change = ifelse(!is.na(ctrl)  & !is.na(dcko), 
                                                            ifelse((ctrl/dcko) > 1, ctrl/dcko, -dcko/ctrl), NA)) %>%
                                     select(cell, short, type_short, change)

reads_stats    <- left_join(reads_stats, reads_stats_f, by = c("cell", "short", "type_short"))
reads_stats_2   <- subset(reads_stats, !is.na(reads_stats$change))
show(reads_stats)
```

Generate table for modification types without dividing by mono- and oligo-. Do not make distinction between short and long tails.
```{r, message=FALSE, warning=FALSE}
reads_stats   <- table %>% group_by(cell, treatment, sample) %>% mutate(all       = n()) %>% ungroup() %>%
                    group_by(cell, treatment, sample, type_short)  %>% mutate(mod_count = n()) %>% ungroup() %>%
                    rowwise() %>% mutate(frequency = 100*mod_count/all)                                %>%
                    select(cell, treatment, sample, type_short, frequency) %>% unique()                   
                    
reads_stats   <- missing_T(reads_stats)

reads_stats   <- reads_stats %>% group_by(cell, treatment, type_short) %>%
                    mutate(mean = mean(frequency), sd = sd(frequency), frequencies=list(frequency)) %>% ungroup()

reads_stats     <- subset(reads_stats , !is.na(reads_stats$sd))
show(reads_stats)

reads_stats_t   <- reads_stats %>% select(cell, treatment, type_short, frequencies) %>% unique() %>% spread(treatment, frequencies)
reads_stats_t   <- reads_stats_t %>% rowwise() %>% 
                                     mutate( pval = ifelse(length(dcko) > 1  & length(ctrl) > 1, 
                                                             t.test(dcko, ctrl, var.equal=TRUE)$p.value, NA)) %>%
                                     mutate( star = ifelse(!is.na(pval), sig(pval),NA)) %>%
                                     select(cell, type_short, pval, star)

reads_stats     <- full_join(reads_stats, reads_stats_t) %>% select(-frequencies)
show(reads_stats)

reads_stats_f   <- reads_stats %>% select(cell, treatment, type_short, mean) %>% unique() %>% spread(treatment, mean)
reads_stats_f   <- reads_stats_f %>% rowwise() %>% 
                                     mutate(change = ifelse(!is.na(ctrl)  & !is.na(dcko), 
                                                            ifelse((ctrl/dcko) > 1, ctrl/dcko, -dcko/ctrl), NA)) %>%
                                     select(cell, type_short, change)

reads_stats     <- left_join(reads_stats, reads_stats_f, by = c("cell", "type_short")) %>% rowwise() %>% mutate(short = "all")
reads_stats     <- subset(reads_stats, !is.na(reads_stats$change))
show(reads_stats)

reads_stats_2 <- bind_rows(reads_stats_2, reads_stats) %>% rename(type = type_short) %>% unique()
```

```{r, message=FALSE, warning=FALSE}
reads_stats_2 <- reads_stats_2 %>% rowwise() %>% mutate(label   = paste(as.character(signif(change, digits = 2)), "X" , "\n", star, sep =""),
                                                        mean_sd = mean + sd)
show(reads_stats_2)
```

Saving source data
```{r, message=FALSE, warning=FALSE}
save <- reads_stats_2 %>% filter(type == "T") %>% select(-type)
show(save)
write.csv(save, "views/final_figures/ED_figure_7_d.csv")
save <- reads_stats_2 %>% filter(type == "C")
write.csv(save, "views/final_figures/ED_figure_9_a.csv")
save <- reads_stats_2 %>% filter(type == "G")
write.csv(save, "views/final_figures/ED_figure_9_b.csv")
```

Make figure legends with number of trancripts and genes for each modification.
```{r, message=FALSE, warning=FALSE}
generate_legend <- function(reader){

    reader           <- reader %>% mutate(cell = factor(cell, levels = c("gv", "liver", "bm", "mefs", "escs")))

    transcripts     <- reader %>% select(cell, treatment,type_short)   %>% 
                             group_by(cell, treatment,type_short) %>% mutate(trans = n())      %>% ungroup() %>%
                             group_by(cell, treatment)            %>% mutate(trans_all = n())  %>% ungroup() %>%
                             unique()

    genes          <- reader %>% select(cell, treatment,type_short, gene) %>% unique() %>% select(cell, treatment,type_short) %>%
                             group_by(cell, treatment,type_short)    %>% mutate(gen     = n()) %>% ungroup() %>%
                             group_by(cell, treatment)               %>% mutate(gen_all = n()) %>% ungroup() %>%
                             unique()

    legend1         <- full_join(transcripts, genes)

    legend1         <- full_join(transcripts, genes) %>% filter(type_short == "T") 

    legend1         <- legend1 %>% select(-trans_all, -gen_all, -type_short) %>% arrange(cell) %>%
                             mutate(print = paste(as.character(trans), " ", "(",  as.character(gen), ")", sep="")) %>%
                             select(-cell, -trans, -gen)  

    show(legend1)
    
    print   <- factor(legend1$print, levels = legend1$print)
    legend1 <- as.data.frame(print) %>% mutate(rep = 0)
    plot    <- ggplot(legend1, aes(x=print)) + theme(axis.text.x = element_text(angle = 90, hjust = 1))
    
    return(plot)
}

```

Save figure legends with numbers of trancripts and genes for each modification. 
```{r, message=FALSE, warning=FALSE}
reads           <- table 
plot <- generate_legend(reads)
plot
ggsave("plots/final_figures/ED_figure_7_d_a_panel.pdf", plot, width=55, height=20, units = "mm", useDingbats=FALSE)

reads           <- table %>% filter(short == "yes") 
plot <- generate_legend(reads)
plot
ggsave("plots/final_figures/ED_figure_7_d_b_panel.pdf", plot, width=55, height=20, units = "mm", useDingbats=FALSE)

reads           <- table %>% filter(short == "no") 
plot <- generate_legend(reads)
plot
ggsave("plots/final_figures/ED_figure_7_d_c_panel.pdf", plot, width=55, height=20, units = "mm", useDingbats=FALSE)
```

```{r}

```

